generator/namespaces.py

```diff
@@ -0,0 +1,88 @@
+"""Generate namespaces.yaml."""
+import tarfile
+import urllib
+import warnings
+from io import BytesIO
+from itertools import groupby
+from operator import itemgetter
+from pathlib import Path
+
+import click
+import requests
+import yaml
+
+
+def _get_first(tuple_):
+    return tuple_[0]
+
+
+def _get_views(uri):
+    with urllib.request.urlopen(uri) as f:
+        tarbytes = BytesIO(f.read())
+    views = {}
```

Comment:
```suggestion
    views = collections.defaultdict(dict)
```
Will let us remove the check on line 28.